### DISCUSSION TITLE
Generate thumbnails+previews for video assets

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#4797] [Feature Requests]
Should support as many formats as possible... and other (non video) formats that I'm forgetting.

http://jsfiddle.net/giu_do/e98tffu6/

**Comment by Nitwel** [comment]
@rijkvanzanten I'll need a bit help on where thumbnail generation is done in the api/app. Or in other words where to implement that.

**Comment by workatease** [comment]
> Should support as many formats as possible... and other (non video) formats that I'm forgetting.
> 
> http://jsfiddle.net/giu_do/e98tffu6/

and where do you think we need to store the saved thumbnail I see some ideas from using ffmpeg library in the API

**Reply by benhaynes** [reply]
I think we should save the video thumb right alongside the other (image) thumbs. I also think that video assets should generate a "video" (MP4) thumbnail that can be used on hover. I'm imagining the video thumbnail processes a small (200x200 contain, with reasonable compression) version that concatenates 10x 1 second clips evenly spaced through the video length. Or just use the full video if its length is less than 10s. Since the image and video thumbnails would have different extensions, there shouldn't be an issue keeping them both in the same thumbs directory.

@workatease — is this a PR you're looking to work on?

**Comment by licitdev** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by al-gerasimov** [comment]
I understand that generating thumbnails for all available types is a complicated task, so could I suggest a workaround to upload an image as a thumbnail? Videos today are more important than still images already, so I think it is an important feature.

**Comment by SvenBudak** [comment]
I agree. We also need video information such as duration, width, height, and especially a preview poster image in almost all of our projects. I would be most happy about an implementation directly in Directus, but I would also be very grateful for a good workaround guide.

**Reply by rijkvanzanten** [reply]
The tricky bit for a 'native' implementation is that we'd have to rely on binaries that may or may  not (most likely not) exist on your system like `ffmpeg` 🤔 That'd make the system requirements quite a bit trickier to manage if you're not using docker
